src/components/Commentable/index.spec.tsx

```diff
@@ -25,14 +48,50 @@ describe(__filename, () => {
   it('is classified as commentable', () => {
     const root = render();
 
+    // This is important because it is instantiating a custom component.
     expect(root).toHaveClassName(styles.commentable);
   });
 
   it('renders children', () => {
     const className = 'child-class';
-    const children = <div className={className} />;
-    const root = render({ children });
+    const root = render({
+      children: (addComment) => (
+        <>
+          {addComment}
+          <div className={className} />
+        </>
+      ),
+    });
 
     expect(root.find(`.${className}`)).toHaveLength(1);
   });
+
+  it('renders AddComment', () => {
+    const addCommentClassName = 'ExampleClass';
+    const fileName = 'manifest.json';
+    const line = 432;
+    const versionId = 1;
+    const root = render({
+      addCommentClassName,
+      children: (addComment) => addComment,
+      fileName,
+      line,
+      versionId,
+    });
+
+    const button = root.find(AddComment);
+    expect(button).toHaveProp('fileName', fileName);
+    expect(button).toHaveProp('line', line);
+    expect(button).toHaveProp('versionId', versionId);
+    expect(button).toHaveClassName(addCommentClassName);
+  });
+
+  it('can add a ref to the outer DOM node', () => {
+    const shellRef = jest.fn();
+
+    // We need `mount()` because `ref` is only used in a DOM environment.
+    const root = renderWithMount({ shellRef });
+
+    expect(shellRef).toHaveBeenCalledWith(root.getDOMNode());
```

Comment:
I take it that this is something that is done automatically when you set the `ref` of `Commentable` to `shellRef`?

Reply:
Yeah. If you delete the `ref` assignment within `Commentable`, this test fails.